### DISCUSSION TITLE
WixSharp and Net 6 or Higer

### DISCUSSION THREAD

**Discussion by gillesnoyer** [oleg-shilo/wixsharp#1451] [Q&A]
Hi Everybody
I use WixSharp to build my project, it's a great tool. Currently my application targets .Net4.8 and I can use it.
I wanted to upgrade to Net6, but nuget packages are not recognized.

My question : Is Wix# compatible with Net 6 or higher?

**Comment by oleg-shilo** [comment]
Technically WixSharp can be used in .NETCore project but WiX will not allow it to host any managed CAs as WiX only supports .NET FX. While there is a good change WiX team may change it in the future, right now this feature is not available. 

Thus you have to use .NET Framework as it is a constraint of the underlying technology (WiX).

**Comment by gillesnoyer** [comment]
Thanks 
by using WhixSharp1.024  I can't include WixSharp  it into a .Net6 project.
by using WixSharp2, I can now include WixSharp  into .Net6 project :) so it is a good news.

As WixSharp2 version is still in dev, I will wait for its official version, but now I know that I can use it into a .Net6 project ( without using managed CA). I note that I need to migrate the project, due to some evolutions of Wix/WixSharp. 

Have you an idea when you will switch to WixSharp2 ?

**Comment by oleg-shilo** [comment]
Sorry, Gilles. It looks like there is some confusion about the matter.

>WixSharp1.024 I can't include WixSharp it into a .Net6 project.

Correct

> by using WixSharp2, I can now include WixSharp into .Net6 project 

I do not know what `WixSharp2`. I have never released such version/edition. Not sure what are you referring to.
WixSharp cannot be used in .NETCore projects. It can be a possibility if WixSharp is recompiled for Core or netstandard but there are no benefits of that as none of .NET Core features can be utilized.

> WixSharp2 version is still in dev,

The same. If you are referring to WiX4 stream of WixSharp then it is released. But the  .NETCore limitations are still in place.

**Comment by oleg-shilo** [comment]
OK, judging by your other post #1452, you are indeed referring to WiX4 edition of WixSharp.

> by using WixSharp2, I can now include WixSharp into .Net6 project 

You still cannot. If you change the target platform to .net6 your wWixSharp assemblies will not be referenced during the build:
![image](https://github.com/oleg-shilo/wixsharp/assets/16729806/6995e8e9-f9e8-4701-b5da-9e03a20f2b40)

The VS template for WixSharp.wix4 uses the same VS project file format as for .NET4 but it still targets .NET4.7.2
![image](https://github.com/oleg-shilo/wixsharp/assets/16729806/c13dd6d4-0aa0-49b1-bb26-9736bf676f2b)

> I can use it into a .Net6 project ( without using managed CA)

Even if WixSharp.wix4 is recompiled for .NET6 and you include it in .NET6 project you would not gain anything. Your code would be the same as for codebase targeting .NET Framework v4.7.1. That's why I am not sure about the intention of integrating WixSharp with .NET Core. There is no single benefit from it until WiX team releases the actual integration for that new CLR.
